### ISSUE TITLE
Diagnostics section

### ISSUE THREAD

**Issue by RAprogramm** [sidebar-nvim/sidebar.nvim#35]
Not showing python diagnostic

**Comment by davysson** [comment]
Do you have a language server configured? Diagnostic information will only be available if you have Neovim lsp configured.

**Comment by RAprogramm** [comment]
yes. all configured. but <no diagnostic> still. all good for lua, but problem with python

**Comment by GustavoKatel** [comment]
hi @RAprogramm we need a bit more info to track this down.

- nvim version
- language server name
- your sidebar setup

**Comment by RAprogramm** [comment]
1. nvim v0.6.1
build type: release
2. i tryed pyright, jedi and pyls.
3.
![sidebar_config](https://user-images.githubusercontent.com/70325462/150506769-5dab1fb0-2f32-466d-91b2-fb922400ec34.png)

**Comment by RAprogramm** [comment]
maybe problem with virtual enviroment?

**Comment by RAprogramm** [comment]
![diagnostics](https://user-images.githubusercontent.com/70325462/150516188-a2069ae9-2cfd-4ff7-8eb5-f174f69a9c39.png)
here is what i am telling about

**Comment by RAprogramm** [comment]
also i find this
![icons_problem](https://user-images.githubusercontent.com/70325462/150516311-5d9aae52-b4d1-4cd4-b208-35e27564e55b.png)
not critical, but sometimes icons looks pretty, and sometimes like this

**Comment by GustavoKatel** [comment]
in `sidebar.lua` you don't have `diagnostics` listed as one of the sections enabled (lines 8-12) and yet you have it in the sidebar.

1- Please make sure to have `diagnostics` listed in the enabled sections.
2- How and where are calling `sidebar.lua` in your dotfiles?

**Comment by RAprogramm** [comment]
No no, when I posted previous screenshot, "diagnostocs" was in my config section. 
.nvim/lua/config/sidebar.lua

**Comment by GustavoKatel** [comment]
hi @RAprogramm did u manage to get it working? if not, do you have your dotfiles available so I can take a look?

**Comment by RAprogramm** [comment]
https://github.com/RAprogramm/neovim-IDE
this is my neovim config.
diagnostic still not working in virtualenvironment python.
in lua works good.

**Comment by GustavoKatel** [comment]
![image](https://user-images.githubusercontent.com/1891977/153773236-2f054c07-8ac1-46cb-ab2f-a3393ae2deed.png)

weird... I'm trying with your setup and I can see the diagnostics section working. I'm using pyright

can you try to force a syntax error in a python file and run `:lua vim.diagnostic.setqflist()`? Let me know the output

**Comment by ghostbuster91** [comment]
I've just run into the same issue but with lua. My diagnostic section is empty.
My sidebar configuration:
```
require("sidebar-nvim").setup({
	["diagnostics"] = {
		icon = "",
	},
	sections = { "datetime", "git", "diagnostics" },
})
```

Results of `:LspInfo`:
```
   Language client log: /home/kghost/.cache/nvim/lsp.log
           Detected filetype:   lua
  
      1 client(s) attached to this buffer:

   Client: sumneko_lua (id: 1, pid: 480206, bufnr: [12])
       filetypes:       lua
       autostart:       true
       root directory:  /home/kghost/workspace/dot-files
       cmd:             lua-language-server

   Configured servers list: vimls, rnix, yamlls, sumneko_lua, tsserver, bashls
```

Results of `:lua vim.diagnostic.setqflist()`:
```
programs/neovim/init.lua|2 col 20-23 warning| Undefined global `vim`.
  1 programs/neovim/init.lua|19 col 1-4 warning| Undefined global `vim`.
  2 programs/neovim/init.lua|23 col 1-4 warning| Undefined global `vim`.
  3 programs/neovim/init.lua|24 col 1-4 warning| Undefined global `vim`.
  4 programs/neovim/init.lua|31 col 3-6 warning| Undefined global `vim`.
  5 programs/neovim/init.lua|53 col 3-6 warning| Undefined global `vim`.
  6 programs/neovim/init.lua|64 col 22-25 warning| Undefined global `vim`.
  7 programs/neovim/init.lua|82 col 11-14 warning| Undefined global `vim`.
  8 programs/neovim/init.lua|83 col 4-7 warning| Undefined global `vim`.
  9 programs/neovim/init.lua|261 col 1-4 warning| Undefined global `vim`.
[Quickfix List] Diagnostics     
```

**Comment by GustavoKatel** [comment]
hi @ghostbuster91 

are you still experiencing this issue with the latest Sidebar updates?

if so, what version of neovim are you using?

**Comment by Suyashtnt** [comment]
It seems that diagnostics only update on file save, while it should update every time e.g enter normal mode

**Comment by GustavoKatel** [comment]
it actually updates using an `autocommand` event called `DiagnosticChanged` (see: https://neovim.io/doc/user/diagnostic.html#DiagnosticChanged)

**Comment by Suyashtnt** [comment]
> it actually updates using an `autocommand` event called `DiagnosticChanged` (see: [neovim.io/doc/user/diagnostic.html#DiagnosticChanged](https://neovim.io/doc/user/diagnostic.html#DiagnosticChanged))

For me it only updated on file save. Strange

**Comment by GustavoKatel** [comment]
> > it actually updates using an `autocommand` event called `DiagnosticChanged` (see: [neovim.io/doc/user/diagnostic.html#DiagnosticChanged](https://neovim.io/doc/user/diagnostic.html#DiagnosticChanged))
> 
> For me it only updated on file save. Strange

🤔 what happens if you call `:lua vim.diagnostic.setqflist({open = true })` before saving? Does it show the expected results?

also what language server and neovim version are you using?

**Comment by Suyashtnt** [comment]
> > > it actually updates using an `autocommand` event called `DiagnosticChanged` (see: [neovim.io/doc/user/diagnostic.html#DiagnosticChanged](https://neovim.io/doc/user/diagnostic.html#DiagnosticChanged))
> > 
> > 
> > For me it only updated on file save. Strange
> 
> thinking what happens if you call `:lua vim.diagnostic.setqflist({open = true })` before saving? Does it show the expected results?
> 
> also what language server and neovim version are you using?

Doesnt show any results. Neovim nightly, LSP is eslint and tsserver

**Comment by Suyashtnt** [comment]
> > > > > it actually updates using an `autocommand` event called `DiagnosticChanged` (see: [neovim.io/doc/user/diagnostic.html#DiagnosticChanged](https://neovim.io/doc/user/diagnostic.html#DiagnosticChanged))
> > > > 
> > > > 
> > > > For me it only updated on file save. Strange
> > > 
> > > 
> > > thinking what happens if you call `:lua vim.diagnostic.setqflist({open = true })` before saving? Does it show the expected results?
> > > also what language server and neovim version are you using?
> > 
> > 
> > Doesnt show any results. Neovim nightly, LSP is eslint and tsserver
> 
> Any chance it's an issue with your setup? I'm also using both eslint and tsserver and they update fine the qflist, but I'm using 0.7

What I mean is setqflist works fine, its just sidebar isnt showing anything

**Comment by GustavoKatel** [comment]
you're saying that you *can* see the qflist populated with the diagnostics?

both `setqflist` and the diagnostics section uses the same source of diagnostics 🤔 

would you be able to create a minimal config that replicates the problem?

**Comment by rodrigoEsquel** [comment]
I also encountered the same issue. I'm also in nightly so maybe there is something with that.

Ran :lua vim.diagnostic.setqflist({open = true }) and everything seems fine
![image](https://github.com/sidebar-nvim/sidebar.nvim/assets/88414096/46ce32ee-8aa0-46aa-875e-b0df23ac460e)

Editing and saving the buffer I was able to make it work once. but just for a while

**Comment by rodrigoEsquel** [comment]
I'll try to do a minimal replication of the error because I'd really like it to get it working. I love the plugin!

**Comment by rodrigoEsquel** [comment]
I was able to get it working by removing one of the two checks that see if the buffers are open in tho diagnostic component.

Particularly, open_bufs[bufnr] was almost always throwing nil

![image](https://github.com/sidebar-nvim/sidebar.nvim/assets/88414096/4764493e-7602-4693-855d-b3a5cf14f4c4)

I don't know what side effects this can bring, tho.
